mailserver/metrics.go

```diff
@@ -1,23 +1,61 @@
-	requestProcessTimer            = metrics.NewRegisteredTimer("mailserver/requestProcessTime", nil)
-	requestProcessNetTimer         = metrics.NewRegisteredTimer("mailserver/requestProcessNetTime", nil)
-	requestsMeter                  = metrics.NewRegisteredMeter("mailserver/requests", nil)
-	requestsBatchedCounter         = metrics.NewRegisteredCounter("mailserver/requestsBatched", nil)
-	requestErrorsCounter           = metrics.NewRegisteredCounter("mailserver/requestErrors", nil)
-	sentEnvelopesMeter             = metrics.NewRegisteredMeter("mailserver/sentEnvelopes", nil)
-	sentEnvelopesSizeMeter         = metrics.NewRegisteredMeter("mailserver/sentEnvelopesSize", nil)
-	archivedMeter                  = metrics.NewRegisteredMeter("mailserver/archivedEnvelopes", nil)
-	archivedSizeMeter              = metrics.NewRegisteredMeter("mailserver/archivedEnvelopesSize", nil)
-	archivedErrorsCounter          = metrics.NewRegisteredCounter("mailserver/archiveErrors", nil)
-	requestValidationErrorsCounter = metrics.NewRegisteredCounter("mailserver/requestValidationErrors", nil)
-	processRequestErrorsCounter    = metrics.NewRegisteredCounter("mailserver/processRequestErrors", nil)
-	historicResponseErrorsCounter  = metrics.NewRegisteredCounter("mailserver/historicResponseErrors", nil)
-	syncRequestsMeter              = metrics.NewRegisteredMeter("mailserver/syncRequests", nil)
-	deliverMailTimer               = metrics.NewRegisteredTimer("mailserver/deliverMailTime", nil)
+	envelopesCounter = prom.NewCounter(prom.CounterOpts{
+		Name: "mailserver_envelopes_total",
+		Help: "Number of envelopes processed.",
+	})
+	deliveryFailuresCounter = prom.NewCounterVec(prom.CounterOpts{
+		Name: "mailserver_delivery_failures_total",
+		Help: "Number of requests that failed processing.",
+	}, []string{"type"})
+	deliveryAttemptsCounter = prom.NewCounter(prom.CounterOpts{
+		Name: "mailserver_delivery_attempts_total",
+		Help: "Number of Whisper envelopes processed.",
+	})
+	requestsBatchedCounter = prom.NewCounter(prom.CounterOpts{
+		Name: "mailserver_requests_batched_total",
+		Help: "Number of processed batched requests.",
+	})
+	requestsInBundlesDuration = prom.NewHistogram(prom.HistogramOpts{
+		Name: "mailserver_requests_bundle_process_duration_seconds",
+		Help: "The time it took to process message bundles.",
+	})
+	syncFailuresCounter = prom.NewCounterVec(prom.CounterOpts{
+		Name: "mailserver_sync_failures_total",
+		Help: "Number of failures processing a sync requests.",
+	}, []string{"type"})
+	syncAttemptsCounter = prom.NewCounter(prom.CounterOpts{
+		Name: "mailserver_sync_attempts_total",
+		Help: "Number of attempts are processing a sync requests.",
+	})
+	sendRawEnvelopeDuration = prom.NewHistogram(prom.HistogramOpts{
+		Name: "mailserver_send_raw_envelope_duration_seconds",
+		Help: "The time it took to send a Whisper envelope.",
+	})
+	sentEnvelopesSizeMeter = prom.NewHistogram(prom.HistogramOpts{
```

Comment:
We should think about the buckets because the default ones do not work well. I don't know the typical sizes of envelopes but it should be e.g. a normal distribution scaled appropriately.

Reply:
We have:
```
mailserver_archived_envelopes_size_bytes_sum 11402
mailserver_archived_envelopes_size_bytes_count 10
```

which means on average the size is 1000. Not sure what's the limit. This could work fine:
```
Buckets: prometheus.LinearBuckets(2**8, 2**20, 10),
```

Reply:
```
func LinearBuckets(start, width float64, count int) []float64
```
So you want to start with `256` bytes and set the high bound at `1 MiB`, makes sense.